functions/go/remove-local-config-resources/main.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/kio/filters"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+//nolint
+func main() {
+	fp := RemoveLocalConfigResourcesConfigProcessor{}
+	cmd := command.Build(&fp, command.StandaloneEnabled, false)
+	if err := cmd.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+func (fp *RemoveLocalConfigResourcesConfigProcessor) Process(resourceList *framework.ResourceList) error {
+	resourceList.Result = &framework.Result{
+		Name: "remove-local-config-resources",
+	}
+
+	items, err := processResources(resourceList)
+	if err != nil {
+		resourceList.Result.Items = getErrorItem(err.Error())
+		return err
+	}
+	resourceList.Result.Items = items
+	return nil
+}
+
+func processResources(resourceList *framework.ResourceList) ([]framework.ResultItem, error) {
+	var items []framework.ResultItem
+	results := []Result{}
+
+	var res []*yaml.RNode
+	for _, node := range resourceList.Items {
+		if node.IsNilOrEmpty() {
+			continue
+		}
+		// only add the resources which are not local configs
+		if strings.ToLower(node.GetAnnotations()[filters.LocalConfigAnnotation]) != "true" {
+			res = append(res, node)
+		} else {
+			itemFilePath := node.GetAnnotations()["internal.config.kubernetes.io/path"]
+			if itemFilePath == "" {
+				itemFilePath = node.GetAnnotations()["config.kubernetes.io/path"]
+			}
+
+			result := Result{
```

Comment:
Instead of creating a `Result` object here. You can directly construct `framework.ResultItem`.
You can eliminate the for loop in line 71. We can reduce the size of this file.

Reply:
Good idea. I've made that change.